packages/spiderable/spiderable_client.js

```diff
@@ -17,14 +17,52 @@
 
 Spiderable._initialSubscriptionsStarted = false;
 
+Spiderable._onReadyHook = new Hook({
+  debugPrintExceptions: "onReadyHook callback"
+});
+
+// register a new onReady hook for validation
+Spiderable.onReady = function (fn) {
+  var self = this;
```

Comment:
I don't find the use of `self` when it's not really an object method to be clearer than just writing `Spiderable`. That said, I think `Spiderable._onReadyHook` could just be `var onReadyHook`.